### ISSUE TITLE
JNI Wrapper

### ISSUE THREAD

**Issue by dpny518** [jcsilva/docker-kaldi-android#15]
Is there some code or instructions on how to call the online2-wav-nnet3-latgen-faster and build your own apk that uses the compiled android kaldi binaries?

https://github.com/ahmetaa/kaldi-jni
https://github.com/keenresearch/KeenASR-Android-PoC
https://github.com/Kidsense/Vocally

**Comment by nanaghartey** [comment]
I've also been trying to accomplish this. I've been using CMUSphinx and pocketsphinx on android and ios but the accuracy is bad compared to Kaldi. I'm sure this is no easy task. KeenResearch has achieved this on ios, android, unity but the licensing costs isn't something indie developers working on small projects can afford :(

**Comment by dpny518** [comment]
@nanaghartey 
https://github.com/rayworks/KaldiDroidDemo

**Comment by nanaghartey** [comment]
@yondu22 Have you tried it and does it work? I realize you opened an issue regarding jni wrappers for the decoder.

**Comment by nanaghartey** [comment]
@yondu22 So I translated the doc from chinese to english and realize it's just about compiling kaldi on android using mac. Have you tried it and does it work? I realize you opened an issue regarding jni wrappers for the decoder.

**Comment by dpny518** [comment]
yes it seems he only did a timer jni library. I have no experience in either C++ or Android Java so I hoping one of you can build this library for me, so just connecting developers on git together

**Comment by dpny518** [comment]
@nanaghartey check this out
https://github.com/alphacep/kaldi-android-demo

**Comment by nanaghartey** [comment]
@yondu22 Awesome! It builds and runs successfully however, the accuracy is poor. I keep getting wrong results. Did you face same issue?
